plugins/modules/v_switch.py

```diff
@@ -0,0 +1,499 @@
+
+    return [
+        server
+        for server in desired_servers
+        if server not in current_ips and server not in current_ids and server not in current_ipv6s
+    ]
+
+
+def set_desired_servers(module, id_):
+    v_switch = get_v_switch(module, id_)
+    changed = False
+
+    if module.params['servers'] is None:
+        return (v_switch, changed)
+
+    servers_to_delete = get_servers_to_delete(v_switch['server'], module.params['servers'])
+    if servers_to_delete:
+        if not module.check_mode:
+            v_switch = delete_servers(module, id_, servers_to_delete)
+        changed = True
+    servers_to_add = get_servers_to_add(v_switch['server'], module.params['servers'])
+    if servers_to_add:
+        if not module.check_mode:
+            v_switch = add_servers(module, id_, servers_to_add)
+        changed = True
+    return (v_switch, changed)
+
+
+def main():
+    argument_spec = dict(
+        vlan=dict(type='int', required=True),
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        servers=dict(type='list', elements='str'),
+        wait=dict(type='bool', default=True),
+        wait_delay=dict(type='int', default=10),
+        timeout=dict(type='int', default=180),
+    )
+    argument_spec.update(ROBOT_DEFAULT_ARGUMENT_SPEC)
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    v_switches, error = fetch_url_json(module, V_SWITCH_BASE_URL, accept_errors=['UNAUTHORIZED'])
+
+    if error:
+        module.fail_json(msg='Please check your current user and password configuration')
+
+    matched_v_switches = [
+        v
+        for v in v_switches
+        if v['name'] == module.params['name'] and v['vlan'] == module.params['vlan']
+    ]
+    non_cancelled_v_switches = [m for m in matched_v_switches if m['cancelled'] is False]
+    result = {'changed': False}
+
+    if len(non_cancelled_v_switches) > 1:
+        module.fail_json(
+            msg='Multiple vSwitches with same name and VLAN ID in non cancelled status. Clean it.'
+        )
+
+    elif len(non_cancelled_v_switches) == 1:
+        id_ = non_cancelled_v_switches[0]['id']
+        v_switch, changed = set_desired_servers(module, id_)
+        if changed:
+            result['changed'] = True
+
+        if module.params['state'] == 'present':
+            result['v_switch'] = v_switch
+        elif module.params['state'] == 'absent':
+            if not module.check_mode:
+                delete_v_switch(module, id_)
+            result['changed'] = True
+        else:
+            # not reachable
+            raise NotImplementedError
+    else:
+        if module.params['state'] == 'present':
+            if not module.check_mode:
+                v_switch = create_v_switch(module)
+            result['changed'] = True
+            if not module.check_mode:
```

Comment:
How about:
```suggestion
            result['changed'] = True
            if not module.check_mode:
                v_switch = create_v_switch(module)
```

Reply:
Changed in 7bdd217fcb174788b0ef20bf64621bb335064960